### PR TITLE
Improve Rust build artifact caching

### PR DESCRIPTION
**Pull request by jaredpalmer**

### CODE REVIEW

File: .github/workflows/build_test_deploy.yml [hunk 1]
```diff
@@ -578,42 +578,19 @@ jobs:
...
-
       # since the repo's dependencies aren't installed we need
       # to install napi globally
       - run: npm i -g @napi-rs/cli@1.2.1
+      - run: npm i -g turbo@next
 
       - name: Build
         if: ${{ steps.binary-cache.outputs.cache-hit != 'true' && steps.docs-change.outputs.DOCS_CHANGE != 'docs only change' }}
-        run: yarn build-native
+        run: turbo run build-native
```
Comment (styfle) [inline .github/workflows/build_test_deploy.yml:588]: Should this be `yarn turbo run build-native` like the others?

File: .github/workflows/pull_request_stats.yml [hunk 2]
```diff
@@ -71,13 +71,13 @@ jobs:
       # since the repo's dependencies aren't installed we need
       # to install napi globally
       - run: npm i -g @napi-rs/cli@1.2.1
+      - run: npm i -g turbo@next
 
       - name: Build
         if: ${{ steps.binary-cache.outputs.cache-hit != 'true' && steps.docs-change.outputs.DOCS_CHANGE != 'docs only change' }}
-        run: yarn build-native
+        run: turbo run build-native
```
Comment (styfle) [inline .github/workflows/pull_request_stats.yml:78]: Same here, should this be `yarn turbo`?

**Comment by jaredpalmer** [comment]
moving to a different PR

**Comment by jaredpalmer** [comment]
Moved to #31464

File: .github/workflows/build_test_deploy.yml (hunk 1 above)
Comment (jaredpalmer) [reply .github/workflows/build_test_deploy.yml:588]: Fixed in https://github.com/vercel/next.js/pull/31463

**Pull request closed without merging**